server/src/main/java/io/crate/expression/operator/LikeOperators.java

```diff
@@ -182,7 +190,9 @@ static final Pattern makePattern(String pattern, CaseSensitivity caseSensitivity
     }
 
     public static boolean matches(String expression, String pattern, CaseSensitivity caseSensitivity) {
-        return makePattern(pattern, caseSensitivity).matcher(expression).matches();
+        return makePattern(pattern, caseSensitivity).matcher(expression).matches()
```

Comment:
I'm not sure this change is correct, because we want consistency.
If there is an index on the field and we use `WildcardQuery`, can we swap the field with the pattern? `SELECT * FROM t where '%a%' like cola` ?

Reply:
scratch that, I've checked, so swapping the args leads to generic function query and it works.